semantic-core/src/Data/Loc.hs

```diff
@@ -10,23 +9,17 @@ import Data.Text (Text, pack)
 import GHC.Stack
 import Source.Span
 
-data Loc = Loc
-  { locPath :: !Text
-  , locSpan :: {-# UNPACK #-} !Span
-  }
+newtype Path = Path { getPath :: Text }
   deriving (Eq, Ord, Show)
 
-interactive :: Loc
-interactive = Loc "<interactive>" (Span (Pos 1 1) (Pos 1 1))
 
-
-here :: HasCallStack => Maybe Loc
+here :: HasCallStack => Maybe (Path, Span)
 here = stackLoc callStack
 
-stackLoc :: CallStack -> Maybe Loc
+stackLoc :: CallStack -> Maybe (Path, Span)
 stackLoc cs = case getCallStack cs of
   (_, srcLoc):_ -> Just (fromGHCSrcLoc srcLoc)
   _             -> Nothing
 
-fromGHCSrcLoc :: SrcLoc -> Loc
-fromGHCSrcLoc SrcLoc{..} = Loc (pack srcLocFile) (Span (Pos srcLocStartLine srcLocStartCol) (Pos srcLocEndLine srcLocEndCol))
+fromGHCSrcLoc :: SrcLoc -> (Path, Span)
+fromGHCSrcLoc SrcLoc{..} = (Path (pack srcLocFile), Span (Pos srcLocStartLine srcLocStartCol) (Pos srcLocEndLine srcLocEndCol))
```

Comment:
My goal is to eventually have these live in `semantic-source` as well, in a `Source.Debug` or similar module.